### DISCUSSION TITLE
Make `BitMap.create` and `BitMap.create_from_image_alpha` static

### DISCUSSION THREAD

**Discussion by dalexeev** [godotengine/godot-proposals#5808] [Engine Core]
I noticed that the methods `Image.create`, `Image.create_from_data` and `ImageTexture.create_from_image` became static (godotengine/godot#60739), while `BitMap.create` and `BitMap.create_from_image_alpha` remained non-static.

Although this trend can be treated differently, Godot is now moving more actively towards static methods (this is especially noticeable in `FileAccess` and `DirAccess`, with their static methods duplicating non-static ones).

And since `Image`, `ImageTexture`, and `BitMap` also belong to the same area, for consistency, we should make the `BitMap.create*` methods also static (and add a non-static alternative for `create_from_image_alpha`, so for `create` there is an alternative in the form of `resize` or `set_bit_rect`, but we can also add `fill`).

![](https://user-images.githubusercontent.com/47700418/202707500-e49844dc-48da-49d7-9095-34790e5713f2.png)

**Comment by dalexeev** [comment]
@KoBeWi You may be interested in this. :)

**Comment by KoBeWi** [comment]
I agree, but this breaks compatibility, so it might be too late to change it.

**Reply by dalexeev** [reply]
As far as I remember, a hard freeze of any backwards incompatible changes (including minor changes and renames) has not yet been announced, only new features and major changes are not considered. That's why I'm asking here since this change looks like a small one.

**Reply by oeleo1** [reply]
Frankly, this is a non issue related to programming comfort. Backwards compatibility primes.

**Reply by Mickeon** [reply]
> As far as I remember, a hard freeze of any backwards incompatible changes (including minor changes and renames) has not yet been announced, only new features and major changes are not considered. That's why I'm asking here since this change looks like a small one.

A small one, plus another small one, plus another small one etc. equals a pretty big set of changes when users have already started porting their projects to 4.0. For that reason, [I've been told the hard way to stop with the renames](https://github.com/godotengine/godot-proposals/issues/5668), and unfortunately this one could be even more confusing to fix because it's a change of method signature.

**Reply by dalexeev** [reply]
This wouldn't be such a problem if major releases were more frequent, Godot 3.0 came out ~5 years ago (January 2018). If major releases are released more frequently in the future and with less drastic changes, then contributors will be less worried about "catching the train", and the transition of users to new major versions will be less painful. (#5445)

This is not a criticism of Godot 4. I and many other users are very excited about the upcoming release of 4.0, but I can not say that there are some users who are "tired of waiting" and do not even believe that the release is coming soon. I just believe that Godot is becoming a mature project and so releases will naturally happen more often and with less pain.

**Comment by dalexeev** [comment]
Let's defer this to Godot 5.0.
